core/src/main/java/org/wildfly/extras/creaper/core/online/HttpModelControllerClient.java

```diff
@@ -0,0 +1,175 @@
+    }
+
+    @Override
+    public ModelNode execute(ModelNode modelNode, OperationMessageHandler handler) throws IOException {
+        return execute(modelNode);
+    }
+
+    @Override
+    public ModelNode execute(Operation operation, OperationMessageHandler handler) throws IOException {
+        return execute(operation);
+    }
+
+    @Override
+    public OperationResponse executeOperation(Operation operation, OperationMessageHandler handler) throws IOException {
+        return OperationResponse.Factory.createSimple(execute(operation));
+    }
+
+    /**
+     * <b>Not supported!</b>
+     */
+    @Override
+    public AsyncFuture<ModelNode> executeAsync(ModelNode modelNode, OperationMessageHandler handler) {
+        throw new UnsupportedOperationException("Asynchronous execution is not supported by " + getClass().getName());
+    }
+
+    /**
+     * <b>Not supported!</b>
+     */
+    @Override
+    public AsyncFuture<ModelNode> executeAsync(Operation operation, OperationMessageHandler handler) {
+        throw new UnsupportedOperationException("Asynchronous execution is not supported by " + getClass().getName());
+    }
+
+    /**
+     * <b>Not supported!</b>
+     */
+    @Override
+    public AsyncFuture<OperationResponse> executeOperationAsync(Operation operation, OperationMessageHandler handler) {
+        throw new UnsupportedOperationException("Asynchronous execution is not supported by " + getClass().getName());
+    }
+
+    @Override
+    public void close() throws IOException {
+    }
+
+    protected HttpPost buildRequest(ModelNode model) throws UnsupportedEncodingException {
+        HttpPost request = new HttpPost(url);
+        request.addHeader("Content-Type", "application/json");
+        request.setEntity(new StringEntity(model.toJSONString(true)));
+        return request;
+    }
+
+    protected DigestScheme getDigestScheme() throws IOException, MalformedChallengeException {
+        // we need default client
+        CloseableHttpClient defaultHttpClient = HttpClients.custom()
+                .setDefaultRequestConfig(requestConfig)
+                .build();
+        final DigestScheme md5Auth = new DigestScheme();
+        HttpPost post = new HttpPost(url);
+        post.addHeader("Content-Type", "application/json");
+
+        CloseableHttpResponse response = defaultHttpClient.execute(post);
+        if (response.getStatusLine().getStatusCode() != HttpStatus.SC_UNAUTHORIZED) {
+            throw new IllegalStateException(String.format("Server responded with %d instead of %d after first request.",
+                    response.getStatusLine().getStatusCode(), HttpStatus.SC_UNAUTHORIZED));
+        }
+        if (!response.containsHeader("WWW-Authenticate")) {
+            throw new IllegalStateException("Missing WWW-Authenticate header in server response after first request.");
+        }
+        md5Auth.processChallenge(response.getHeaders("WWW-Authenticate")[0]);
+        defaultHttpClient.close();
```

Comment:
This won't execute if an exception is thrown above. Should be in a `finally` block.